feature_engine/prediction/target_mean_prediction.py

```diff
@@ -0,0 +1,244 @@
+            Values are the mean values associated with the corresponding encoded or discretised bin
+
+        """
+        # NOTES:
+        # - X needs to be a dataframe to be compatible w/ the BaseEncoder()
+        # - X needs to match the shape of the dataframe used in fit()
+
+        # check if is pandas series w/ a name that matches
+        #
+        # if not isinstance(X, pd.Series):
+        #     raise TypeError("fit() method only accepts pandas series.")
+
+        if variable_name not in (self.variables_):
+            raise ValueError("Series name does not match the dataframe features that were used to "
+                             "fit the predictor. The variables that were fitted to the predictor: "
+                             f"{self.varialbes_}.")
+
+        # use fitted encoder to return the corresponding mean for each category
+        if variable_name in self.variables_categorical_:
+            X_prediction = self.encoder.transform(X)
+
+        # transform the numerical to the appropriate discretised bin.
+        # replace the bin index w/ the corresponding mean value.
+        else:
+            X_transformed = self.discretiser.transform(X)
+            X_prediction = X_transformed.apply(lambda bin_idx: self.disc_mean_dict_[bin_idx])
+
+        return X_prediction[variable_name]
+
+    def _make_numerical_pipeline(self):
+
+        if self.strategy == "equal-width":
+            discretiser = EqualWidthDiscretiser(
+                bins=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+        else:
+            discretiser = EqualFrequencyDiscretiser(
+                q=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+
+        encoder = MeanEncoder(variables=self.variables_numerical_)
+
+        _pipeline_numerical = Pipeline(
+            [
+                ("discretisation": discretiser),
+                ("encoder": encoder),
+            ]
+        )
+
+        return _pipeline_numerical
+
+    def _make_categorical_pipeline(self):
+
+        return MeanEncoder(variables=self.variables_categorical_)
+
+    def _make_combined_pipeline(self):
+
+        if self.strategy == "equal-width":
+            discretiser = EqualWidthDiscretiser(
+                bins=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+        else:
+            discretiser = EqualFrequencyDiscretiser(
+                q=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+
+        encoder_num = MeanEncoder(variables=self.variables_numerical_)
+        encoder_cat = MeanEncoder(variables=self.variables_categorical__)
+
+        _pipeline_combined = Pipeline(
+            [
+                ("discretisation": discretiser),
+                ("encoder_num": encoder_num),
+                ("encoder_cat": encoder_cat),
+            ]
+        )
+
+        return _pipeline_combined
+
+
```

Comment:
need to add the score() method, pls have a look at the classifiermixin and regressionmixin to see what sklearn are doing :p